ark/mibi/tiling_utils.py

```diff
@@ -197,12 +320,87 @@ def set_tiling_params(fov_list_path, moly_path):
 
         region_rand.append(randomize)
 
+
+def set_tiling_params(fov_list_path, moly_path, tma=False):
+    """Given a file specifying fov regions, set the MIBI tiling parameters
+
+    User inputs will be required for many values. Also returns moly_path data.
+
+    Args:
+        fov_list_path (str):
+            Path to the JSON file containing the fovs used to define each tiled region
+        moly_path (str):
+            Path to the JSON moly point file, needed to separate fovs
+        tma (bool):
+            Whether the data in fov_list_path is in TMA format or not
+
+    Returns:
+        tuple:
+            Contains:
+
+            - A dict containing the tiling parameters for each fov
+            - A dict defining the moly points to insert if specified
+    """
+
+    # file path validation
+    if not os.path.exists(fov_list_path):
+        raise FileNotFoundError("FOV region file %s does not exist" % fov_list_path)
+
+    if not os.path.exists(moly_path):
+        raise FileNotFoundError("Moly point file %s does not exist" % moly_path)
+
+    # read in the fov list data
+    with open(fov_list_path, 'r') as flf:
+        fov_tile_info = json.load(flf)
+
+    # read in the moly point data
+    with open(moly_path, 'r') as mpf:
+        moly_point = json.load(mpf)
+
+    # define the parameter dict to return
+    tiling_params = {}
+
+    # retrieve the format version
+    tiling_params['fovFormatVersion'] = fov_tile_info['fovFormatVersion']
+
+    # define lists to hold the starting x and y coordinates for each region
+    region_start_x = []
+    region_start_y = []
+
+    # define lists to hold the number of fovs along each axis
+    fov_num_x = []
+    fov_num_y = []
+
+    # define lists to hold the size of each fov
+    x_fov_size = []
+    y_fov_size = []
+
+    # define a list to determine if the fovs should be randomly ordered
+    region_rand = []
+
+    # if TMA is set we also need to define ending coordinates and intervals for each region
+    x_intervals = [] if tma else None
+    y_intervals = [] if tma else None
```

Comment:
Noticing this pattern of defining empty lists before filling them in `read_tma_region_input()` or the other function.  If these arguments will always be empty lists, could we just return them in a dictionary/tuple/NamedTuple and then just pass that data structure to the other functions?